account_ux/wizards/account_change_currency.py

```diff
@@ -43,37 +43,34 @@ def get_move(self):
         default='currency'
     )
 
-    @api.onchange('currency_to_id')
-    def onchange_currency(self):
-        if not self.currency_to_id:
-            self.currency_rate = False
-        else:
-            currency = self.currency_from_id.with_context(
-                )
-            self.currency_rate = currency._convert(
-                1.0, self.currency_to_id, self.move_id.company_id,
-                date=self.move_id.date or
-                fields.Date.context_today(self))
-
     def change_currency(self):
         self.ensure_one()
         if self.change_type == 'currency':
-            self.currency_rate = 1
-        message = _("Currency changed from %s to %s with rate %s") % (
-            self.move_id.currency_id.name, self.currency_to_id.name,
-            self.currency_rate)
+            currency = self.currency_to_id
+            if currency != self.move_id.company_id.currency_id:
+                self.currency_rate = currency._convert(
+                    1.0, self.currency_from_id, self.move_id.company_id,
+                    date=self.move_id.date or
+                    fields.Date.context_today(self))
+            else:
+                self.currency_rate = 1
+                self.move_id.l10n_ar_currency_rate = '0.0'
```

Comment:
```suggestion
                self.move_id.l10n_ar_currency_rate = 0.0
```
en este caso el valor seria el valor del numero flotanto, no seria necesario pasarlo en una cadena '0.0'